product-service/src/functions/createProduct/handler.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  APIGatewayProxyEventHandler,
+  formatBadRequestResponse,
+  formatJSONResponse,
+  formatServerErrorResponse
+} from '@libs/api-gateway';
+import { randomUUID } from 'crypto';
+import { addProductToDB, addStockToDB, getProducts } from '@libs/data-service';
+import { CreateProductParams } from 'src/types/api-types';
+
+const validateParams = (data: CreateProductParams) => {
+  const requiredFields: (keyof CreateProductParams)[] = ['title', 'description', 'image', 'price'];
+  const missingFields = requiredFields.filter((field) => !(field in data));
+
+  if (missingFields.length > 0) {
+    const errorFields = missingFields.join(', ');
+
+    const error = `Error: ${errorFields} ${missingFields.length > 1 ? 'are' : 'is'} not provided`;
+
+    return error;
+  } else {
+    return null;
+  }
+};
+
+export const main: APIGatewayProxyEventHandler = async (event) => {
+  try {
+    const productId = randomUUID();
+
+    const data = JSON.parse(event.body || '{}') as unknown as CreateProductParams;
```

Comment:
**Suggestion**:  The best way create middleware  and add bodyParser (for example from middy) there.